src/main/java/org/folio/linked/data/mapper/dto/ResourceDtoMapper.java

```diff
@@ -28,6 +28,9 @@
 
 @Log4j2
 @Mapper(componentModel = SPRING, imports = {Collectors.class, Arrays.class, ResourceTypeEntity.class})
+// We cannot use constructor injection in the subclass due to https://github.com/mapstruct/mapstruct/issues/2257
+// so, we use field injection here.
+@SuppressWarnings("java:S6813")
 public abstract class ResourceDtoMapper {
 
   @Autowired
```

Comment:
in the issue discussion that you have specified here I saw such suggestion
```java
@Setter(onMethod_ = {@Autowired})
private SingleResourceMapper singleResourceMapper;
```
looks like Sonar doesn't complain on such code